core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java

```diff
@@ -438,6 +439,10 @@ protected void serialize(File outputFile, StoreValue<T> storeValue) throws Objec
             {
                 DeserializationPostInitialisable.Implementation.init(storedValue.getValue(), muleContext);
             }
+            if (storedValue.getValue() instanceof WrapperOrderEvent)
```

Comment:
this is smelly code. In any case, `WrapperOrderEvent` should implement `DeserializationPostInitialisable`